src/test/java/teammates/test/cases/logic/InstructorsLogicTest.java

```diff
@@ -728,4 +730,25 @@ private void verifySameInstructor(InstructorAttributes instructor1, InstructorAt
         assertEquals(instructor1.email, instructor2.email);
     }
 
+    private void testGetCoOwnersForCourse() {
+        ______TS("Verify co-owner status of generated co-owners list");
+        String courseId = "idOfTypicalCourse1";
+        List<InstructorAttributes> generatedCoOwners = instructorsLogic.getCoOwnersForCourse(courseId);
+        for (InstructorAttributes generatedCoOwner : generatedCoOwners) {
+            assertTrue(generatedCoOwner.hasCoownerPrivileges());
+        }
+
+        ______TS("Verify all co-owners present in generated co-owners list");
+
+        List<InstructorAttributes> instructors = instructorsLogic.getInstructorsForCourse(courseId);
```

Comment:
Instead of generating your expected results from instructorsLogic (something you are testing), I think it may be better to generate it from the dataBundle.